File: 2000/01/main.rs
use std::env;

mod util;

struct Day0Solution {
  input: Vec<String>
}

impl util::Solution for Day0Solution {
  fn part1_result(&self) -> String {
    self.input[0].to_string()
  }

  fn part2_result(&self) -> String {
    self.input[1].to_string()
  }
}

fn main() {
  let input = vec!["Hello".to_string(), "World!".to_string()];
  let solution = Day0Solution { input: input };
  let executor = util::Executor::new(&solution as &util::Solution);
  let args = env::args().collect();
  executor.run(args);
}

File: 2000/01/Main.java
import java.util.ArrayList;

import tcollier.Executor;
import tcollier.Solution;

class Day0Solution implements Solution<String> {
  public String part1Answer(ArrayList<String> input) {
    return input.get(0);
  }

  public String part2Answer(ArrayList<String> input) {
    return input.get(1);
  }
}

class Main {
  public static void main(String[] args) {
    ArrayList<String> input = new ArrayList<String>();
    input.add("Hello");
    input.add("World!");
    Executor<String> executor = new Executor<String>(new Day0Solution(), input);
    executor.run(args);
  }
}
